src/allocation_engine/interval_tree.rs

```diff
@@ -166,48 +44,66 @@ pub enum NodeState {
 }
 
 impl NodeState {
+    /// Checks if the node is free.
     #[allow(dead_code)]
-    pub(crate) fn is_free(&self) -> bool {
+    pub fn is_free(&self) -> bool {
         *self == NodeState::Free
     }
 }
 
 /// Internal tree node to implement interval tree.
 #[derive(Clone, Debug, PartialEq, PartialOrd, Eq, Ord)]
-pub struct InnerNode {
+struct InnerNode {
     /// Interval handled by this node.
-    key: Range,
+    pub key: Range,
     /// NodeState, can be Free or Allocated.
-    node_state: NodeState,
+    pub node_state: NodeState,
     /// Optional left child of current node.
-    left: Option<Box<InnerNode>>,
+    left: Option<Node>,
     /// Optional right child of current node.
-    right: Option<Box<InnerNode>>,
+    right: Option<Node>,
     /// Cached height of the node.
     height: u64,
 }
 
-impl InnerNode {
-    /// Creates a new InnerNode object.
+/// Wrapper structure over InnerNode, implements all methods needed to manage
```

Comment:
`InnerNode` is a private structure so we shouldn't mention it in public comments. We can simply say: "Wrapper over a node that implements the required functionality for managing an address space as interval trees".